build/containers/Dockerfile.final

```diff
@@ -0,0 +1,139 @@
+# ####################################
+
+FROM base as build-py-release
+
+# Setup build environment
+ENV PYTHONFAULTHANDLER=1 \
+  PYTHONUNBUFFERED=1 \
+  PYTHONHASHSEED=random \
+  PIP_NO_CACHE_DIR=1 \
+  PIP_DISABLE_PIP_VERSION_CHECK=1 \
+  PIP_DEFAULT_TIMEOUT=100 \
+  POETRY_VERSION=1.2.1
+
+# Install all dependencies. Python venv 
+RUN pip install "poetry==$POETRY_VERSION"
+RUN apt-get update && apt-get install -y make gcc python3-dev && apt-get install -y --reinstall build-essential && python -m venv /venv
+
+# setup python config and install
+COPY pyproject.toml poetry.lock ./
+RUN poetry export --without-hashes -f requirements.txt | /venv/bin/pip install -r /dev/stdin 
+
+# ####################################
+## 3.1 bring js dependencies and code
+# ####################################
+
+FROM base as build-js-release
+
+WORKDIR /app
+
+# Install Node dependencies to /app/node_modules/
+COPY package.json package-lock.json tailwind.config.js /app/
+RUN npm install 
+
+# Copy source to temporary directory. This stage is a disposal layer. Removing /tmp/ambuda is non-issue
+COPY ./ambuda/ /app/ambuda/
+
+# Copy css
+RUN npx tailwindcss -i /app/ambuda/static/css/style.css -o /app/ambuda/static/gen/style.css --minify
+
+# Copy js
+RUN npx esbuild /app/ambuda/static/js/main.js --outfile=/app/ambuda/static/gen/main.js --bundle --minify
+
+RUN echo ">>>>>>>>>> JS code build succeeded! <<<<<<<<<<<<<"
+
+# ####################################
+## 4. Copy Ambuda codes
+# ####################################
+
+FROM base as build-ambuda 
+
+ARG BUILD_TYPE=release
+
+RUN echo ">>>>>>>>>> Building FINAL ${BUILD_TYPE}<<<<<<<<<<<<<"
+COPY --from=build-py-release /venv /venv
+COPY --from=build-js-release /app /app
+
+# Setup build environment
+ENV PYTHONFAULTHANDLER=1 \
+  PYTHONUNBUFFERED=1 \
+  PYTHONHASHSEED=random \
+  PIP_NO_CACHE_DIR=1 \
+  PIP_DISABLE_PIP_VERSION_CHECK=1 \
+  PIP_DEFAULT_TIMEOUT=100 \
+  POETRY_VERSION=1.2.1 \
+  PYTHONPATH=${PYTHONPATH}:/app/
+
+
+# Install all dependencies. Python venv 
+RUN apt-get update && apt-get install -y git
+
+### TODO: Move Translations to the bottom image
+# Install all i18n files
+RUN . /venv/bin/activate && python /app/ambuda/scripts/fetch_i18n_files.py && pybabel compile -d /app/ambuda/translations -f
+
+# specific files
+COPY config.py /app/
+COPY wsgi.py /app/
+COPY alembic.ini /app/
+COPY migrations/ /app/migrations/
+COPY scripts/ /app/scripts/
+RUN echo ">>>>>>>>>> Ambuda application build succeeded! <<<<<<<<<<<<<"
+
+# ####################################
+## 5. Finalize the image
+# ####################################
+
+FROM base as final
+
+ARG BUILD_TYPE=release
+WORKDIR /app
+
+RUN echo ">>>>>>>>>> Building FINAL ${BUILD_TYPE}<<<<<<<<<<<<<"
+COPY --from=build-ambuda /venv /venv
+COPY --from=build-ambuda /app /app
+
+ENV PATH="/venv/bin:$PATH" \
+   FLASK_ENV=development \
+   FLASK_UPLOAD_FOLDER=/app/data/file-uploads \
+   SQLALCHEMY_DATABASE_URI=sqlite:///database.db \
```

Comment:
Why do we need to hardcode these envs? Aren't they passed in through docker-compose?

Reply:
I agree. I didn't get to test without these env variables. These ENV vars are not necessary.